Gateway & Dashboard 5.11.0 RNs.mdx (#1018)
TykTechnologies/tyk-docs: developer-support/release-notes/dashboard.mdx

---

## 5.11 Release Notes 

### 5.11.0 Release Notes 

#### Release Date xx December 2025

#### Release Highlights

> 

Reviewer: ```suggestion

Tyk 5.11 delivers security enhancements, streamlined developer workflows, and deeper operational visibility - empowering teams to scale their API programs with confidence and efficiency.

**Strengthened API Security & Authentication**
This release advances our security foundation with enhanced JWT authentication capabilities. Teams can now leverage scope-to-policy mapping without requiring default policies, while new support for nested claims enables more granular policy and subject identification. We've also added IP spoofing protection through configurable depth selection in X-Forwarded-For headers, and introduced a new Dashboard API endpoint that allows admins to clear JWKS cache across all connected Gateways.

**Improved Certificate Management & Visibility**
Certificate management becomes more proactive with enhanced visibility showing which APIs are impacted by each certificate, plus highlighting certificates approaching expiry to prevent service disruptions before they occur.

**Enhanced Developer Experience**
API integration gets simpler with automatically populated OpenAPI descriptions that include all accessible URLs based on deployment targets and version identifiers. This reduces integration errors and accelerates time-to-value for API consumers.

**Advanced Operational Control**
Operations teams gain greater flexibility with the ability to temporarily remove targets from upstream load balancers during maintenance windows. Enhanced observability comes through OTel trace propagation to custom gRPC plugins, trace ID inclusion in API traffic logs, and dedicated Gateway latency metrics alongside upstream measurements.

**Stability & User Experience Improvements**
This release also includes important fixes for memory exhaustion issues when Redis logging is enabled, improved consistency between Dashboard and Gateway validation for JWT configurations, and streamlined policy management with standardized Policy ID usage across the platform. Multiple UI improvements enhance the user experience including fixes for API filtering, certificate search functionality, Quick Start onboarding flow, and various dashboard display issues - ensuring a smoother workflow for API management and configuration tasks.

These enhancements collectively strengthen Tyk's position as the platform of choice for organizations requiring enterprise-scale API management with robust security, operational excellence, and developer productivity.
```

Please note that the Tyk Helm Charts are configured to install the LTS version of Tyk Dashboard. You will need to modify them to install v5.11.0.

#### Changelog
<a id="Changelog-v5.11.0"></a>

##### Added

<AccordionGroup>

<Accordion title='Improved Flexibility for Key Expiration Settings'>
We've improved key expiration flexibility in the Tyk Dashboard by adding support for custom expiration periods using human-readable time formats (e.g., "5h,34m,30s"). Users can now input custom timeframes supporting months, weeks, days, hours, and seconds in addition to the existing predefined options, eliminating the previous restriction and removing the need for API calls or manual key revocation for custom expiration times.
</Accordion>

<Accordion title='Open Policy Agent Support for Distinguishing Tyk OAS APIs'>
Added three new OPA helper functions (`isTykOas`, `isTykClassic`, and `isTykStreams`) that can be used in OPA rules to distinguish between different API types, ensuring that logic is only applied to the appropriate API types.
</Accordion>

<Accordion title='Added API Endpoint for Certificate Dependencies'>

> Introduced a new Dashboard API endpoint `GET /certs/{certificateId}/apis` that efficiently retrieves all APIs linked to a specific certificate. The endpoint returns API names, IDs, and usage types (client, upstream, server) with proper pagination support, enabling the frontend to display certificate impact information without expensive client-side filtering across all APIs.

Reviewer: ```suggestion
Introduced a new Dashboard API endpoint `GET /certs/{certificateId}/apis` that retrieves all APIs linked to a specific certificate. The endpoint returns API names, IDs, and how they are used by each API (client, upstream, server) with proper pagination support, enabling the frontend to display certificate impact information without expensive client-side filtering across all APIs.
```

</Accordion>

<Accordion title='Improvements in the Dashboard UI'>
- **Proactive Certificate Expiry Notifications in Tyk Dashboard**: Added banner notification to inform the user if the Dashboard has received a certificate expiring soon/expired event from the Gateway. Such certificates are now highlighted on the certificates screen.
- **Improved Automatic Configuration Options for API Creation**: Enhanced the user experience when importing OpenAPI descriptions by replacing checkboxes with toggle switches and improving the wording of configuration options. The new interface provides clearer descriptions for Request Validation, Mock Responses, Authentication Detection, and Path Restrictions.

> - **Improve API Designer UI to clearly display versioned API URLs and access methods**: The Tyk OAS API Designer now displays additional information for the base path URLs where published APIs can be accessed, including versioning information, custom domains, and distributed data plane gateways (when declared in the Dashboard's `edge_endpoints` configuration). To accompany and support this enhanced visibility, Tyk will now add all base paths to the `servers` section in the OpenAPI description. We have introduced a new Dashboard API endpoint (`GET /api/apis/oas/{apiID}/urls`) that returns the list of base path URLs extracted from the Tyk OAS API definition presented in a structured format including detailed components (protocol, domain, port, listen-path).

Reviewer: I would extract this one to a separate entry - and turn it around to start from the changes made to OpenAPI description first. Perhaps something like:

Title: Improved visibility of API base paths

Content: Tyk has always added the [base path](api-management/gateway-config-managing-oas#api-base-path) to the `servers` section of an API's OpenAPI description. Until now, this has been constructed from the address of the local Gateway with the unique listen path for the API. This is not always accessible, for example for child API versions configured for internal access.

Now Tyk will calculate all the URLs where the API can be accessed, adding these to the `servers` section of the OpenAPI description. This takes into account the versioning identifier if URL location is in use (e.g. <gateway_url>/<listen_path>/v1) and, where `edge_endpoints` are defined in the Dashboard config, the data plane Gateways where the API is deployed.

We have introduced a new Dashboard API endpoint (`GET /api/apis/oas/{apiID}/urls`) that returns a list of all of these base paths, presented in a structured format including detailed components (protocol, domain, port, listen-path, versioning identifier).

The API Designer uses the data from this endpoint to display all known API URLs and also indicates how to select the specific API version (if applicable) using URL, query parameter or header.

- **Enhanced Certificate Management with Impact Visibility**: Introduced a new Certificate Details page that displays user-friendly certificate metadata with a toggle between formatted and raw JSON views. The page now shows certificate status indicators (valid, expiring soon, expired) and lists all APIs using the certificate with clickable navigation to their API Designer pages.
</Accordion>

<Accordion title='Removed Default Policy Requirement for JWT Scope-to-Policy Mapping'>
We’ve removed the need to supply a default policy when using scope-to-policy mapping with JWT Authentication. Now, if you enable scope-to-policy mapping by configuring `scopes.claimName`, you do not need to provide a policy ID in `defaultPolicies`. If a request does not contain any valid scopes, it will be rejected with `HTTP 403 Forbidden` (default deny). You can still provide a default policy if you require a different behaviour.
</Accordion>

> <Accordion title='Add Trace ID to Traffic Logs for Improved Observability'>

Reviewer: This is a Gateway change (the Dashboard PR was just for integration tests). Please remove.

When OpenTelemetry is enabled, the Trace ID allocated to each request is tagged in traffic logs as `trace-id-{traceID}` and also exposed in `$tyk_context.tyk_trace_id`. This makes it easier to correlate traffic logs with OTel traces in observability platforms and also allows transformation middleware and custom plugins to correlate logs with traces.
</Accordion>

> <Accordion title='Added support for certificate-to-token binding with Auth Token authentication when using static mTLS'>

Reviewer: This is not included in 5.11.0
